File: Medium/382. Linked List Random Node/LinkedListRandomNode.java
/*
LeetCode Medium: 382. Linked List Random Node
https://leetcode.com/problems/linked-list-random-node/

Check LinkedListRandomNode.cpp for brief solution explanation.
*/
class Solution {
    private ArrayList<Integer>arr;
    public Solution(ListNode head) {
        arr = new ArrayList<Integer>();
        while(head!=null){
            arr.add(head.val);
            head = head.next;
        }
    }
    
    public int getRandom() {
        return arr.get((int)(Math.random()*arr.size()));
    }
}

File: Medium/382. Linked List Random Node/LinkedListRandomNode.py
'''
LeetCode Medium: 382. Linked List Random Node
https://leetcode.com/problems/linked-list-random-node/

Check LinkedListRandomNode.cpp for brief solution explanation.
'''
class Solution:

    def __init__(self, head: Optional[ListNode]):
        self.arr = []
        while head is not None:
            self.arr.append(head.val)
            head = head.next

    def getRandom(self) -> int:
        return self.arr[randint(0, len(self.arr)-1)]